clang/include/clang/Basic/AttrDocs.td

```diff
@@ -3124,6 +3124,26 @@ so we have this attribute in sycl_detail namespace.
   }];
 }
 
+def SYCLHostPipeAttrDocs : Documentation {
+  let Category = DocCatType;
+  let Heading = "__sycl_detail__::host_pipe";
+  let Content = [{
+This attribute is part of support for SYCL host_pipe feature.
+Global or static variables of type decorated with this attribute have
+`sycl-unique-id`, an LLVM IR attribute, added to the definition of each such
+variable, which provides a unique string identifier using
+__builtin_sycl_unique_stable_id.
+We do not intend to support this as a general attribute that user code can use,
+so we have this attribute in sycl_detail namespace.
+
+.. code-block:: c++
+  struct
+  [[__sycl_detail__::host_pipe]] __pipeType {}
+
+  __pipeType __pipe;
+  }];
+}
```

Comment:
@elizabethandrews - Now that we have `[[__sycl_detail__::sycl_type(...)]]`, could this be accomplished using it? If so, we may consider doing the same for `[[__sycl_detail__::device_global]]`

Reply:
I think we can use `sycl_type ` here. I have a JIRA to replace `device_global`  to use `sycl_type` as well. Just haven't gotten around to it.

Reply:
@elizabethandrews @steffenlarsen Are there examples or documentation for `sycl_type`? I took a quick look in the repo and it's not clear to me how I should use it here.

Reply:
@rho180 you can see examples of use here - https://github.com/intel/llvm/pull/6674. You just need to add `host_pipe` to the definition

Reply:
@elizabethandrews @steffenlarsen Updated. Though I know it's not best practice, I copied the static function isSyclType() from SemaSYCL.cpp to do the type check in CodeGenModule. Let me know if there's a common place where this can be put and I can make that change as well.